tests/system/model_monitoring/test_model_monitoring_api.py

```diff
@@ -0,0 +1,554 @@
+        assert len(endpoints_intersect) == 5
+
+    def test_list_endpoints_filter(self):
+        auth_info = self._get_auth_info()
+        access_key = auth_info.data_session
+        for i in range(5):
+            endpoint_details = self._mock_random_endpoint()
+
+            if i < 1:
+                endpoint_details.spec.model = "filterme"
+
+            if i < 2:
+                endpoint_details.spec.function_uri = "test/filterme"
+
+            if i < 4:
+                endpoint_details.metadata.labels = {"filtermex": "1", "filtermey": "2"}
+
+            mlrun.api.crud.ModelEndpoints().write_endpoint_to_kv(
+                access_key, endpoint_details
+            )
+
+        filter_model = mlrun.api.crud.ModelEndpoints().list_endpoints(
+            auth_info, self.project_name, model="filterme"
+        )
+        assert len(filter_model.endpoints) == 1
+
+        filter_labels = mlrun.api.crud.ModelEndpoints().list_endpoints(
+            auth_info, self.project_name, labels=["filtermex=1"]
+        )
+        assert len(filter_labels.endpoints) == 4
+
+        filter_labels = mlrun.api.crud.ModelEndpoints().list_endpoints(
+            auth_info, self.project_name, labels=["filtermex=1", "filtermey=2"],
+        )
+        assert len(filter_labels.endpoints) == 4
+
+        filter_labels = mlrun.api.crud.ModelEndpoints().list_endpoints(
+            auth_info, self.project_name, labels=["filtermey=2"]
+        )
+        assert len(filter_labels.endpoints) == 4
+
+    def test_get_endpoint_metrics(self):
+        auth_info = self._get_auth_info()
+        access_key = auth_info.data_session
+
+        path = config.model_endpoint_monitoring.store_prefixes.default.format(
+            project=self.project_name, kind=mlrun.api.crud.ModelEndpoints().EVENTS
+        )
+        _, container, path = parse_model_endpoint_store_prefix(path)
+
+        frames = get_frames_client(
+            token=access_key, container=container, address=config.v3io_framesd,
+        )
+
+        start = datetime.utcnow()
+
+        for i in range(5):
+            endpoint = self._mock_random_endpoint()
+            mlrun.api.crud.ModelEndpoints().write_endpoint_to_kv(access_key, endpoint)
+            frames.create(backend="tsdb", table=path, rate="10/m", if_exists=1)
+
+            total = 0
+
+            dfs = []
+
+            for i in range(10):
+                count = randint(1, 10)
+                total += count
+                data = {
+                    "predictions_per_second_count_1s": count,
+                    "endpoint_id": endpoint.metadata.uid,
+                    "timestamp": start - timedelta(minutes=10 - i),
+                }
+                df = pd.DataFrame(data=[data])
+                dfs.append(df)
+
+            frames.write(
+                backend="tsdb",
+                table=path,
+                dfs=dfs,
+                index_cols=["timestamp", "endpoint_id"],
+            )
```

Comment:
Maybe I'm misunderstanding, but it seems like you're mocking predictions metrics here
I don't see a reason why not bring up a real serving function, send it requests, and then the metrics should be created automatically and no reason to mock them, that's the idea of system tests, test the whole flow end to end as the real world user would use it